libcoap/src/protocol.rs

```diff
@@ -233,6 +234,10 @@ pub enum CoapContentFormat {
     SensMlXml = COAP_MEDIATYPE_APPLICATION_SENSML_XML as u16,
     ApplicationXml = COAP_MEDIATYPE_APPLICATION_XML as u16,
     TextPlain = COAP_MEDIATYPE_TEXT_PLAIN as u16,
+    AceCbor = COAP_MEDIATYPE_APPLICATION_ACE_CBOR as u16,
+    CoapGroupJson = COAP_MEDIATYPE_APPLICATION_COAP_GROUP_JSON as u16,
+    MbCborSeq = COAP_MEDIATYPE_APPLICATION_MB_CBOR_SEQ as u16,
+    OsCore = COAP_MEDIATYPE_APPLICATION_OSCORE as u16,
     Other,
```

Comment:
Hmm, not really related to this PR, but could you also have `Other` accept the respective Content-Format value here?

Reply:
I can try, but it would prevent us from casting instances of this enum to their respective representation (see https://doc.rust-lang.org/reference/items/enumerations.html#casting), which might break some other code.

Reply:
For the reason stated above, this is unfortunately not possible without making some larger changes that i would prefer to make in a separate PR.

Maybe we could use an approach similar to the one `coset` uses: https://docs.rs/coset/latest/coset/enum.RegisteredLabelWithPrivate.html

Reply:
> For the reason stated above, this is unfortunately not possible without making some larger changes that i would prefer to make in a separate PR.
> 
> Maybe we could use an approach similar to the one `coset` uses: https://docs.rs/coset/latest/coset/enum.RegisteredLabelWithPrivate.html

That sounds good to me, thank you for the consideration :)

Reply:
I have created #16 to address this.